### PR TITLE
 feat: create the components inside the beagle android

### PR DESCRIPTION
**Pull request by uziasferreirazup**
## Note
Create the components inside in android

BREAKING CHANGE: change components import
BREAKING CHANGE: remove components stack, horizontal and vertical

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/layout/StackTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.components.layout
+
```
Comment (tuliopereirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/components/layout/StackTest.kt:18]: Remove this unnecessary space.

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/utils/CanvasExtensionsKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.components.utils
+
+import org.junit.Assert.*
+
+class CanvasExtensionsKtTest
```
Comment (tuliopereirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/components/utils/CanvasExtensionsKtTest.kt]: Does exists test for this class ?

File: schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/layout/Screen.kt [hunk 3]
```diff
@@ -111,11 +111,11 @@ data class NavigationBar(
  * @param screenAnalyticsEvent send event when screen appear/disappear
  *
  */
-data class Screen(
-    val identifier: String? = null,
-    val safeArea: SafeArea? = null,
-    val navigationBar: NavigationBar? = null,
-    val child: ServerDrivenComponent,
-    val style: Style? = null,
+open class Screen(
```
Comment (tuliopereirazup) [inline schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/layout/Screen.kt:114]: Did you miss to delete this `open` or is a requirement?

File: maven-publish.gradle [hunk 4]
```diff
@@ -80,13 +80,7 @@ if (project.hasProperty("android")) { // Android libraries
         source = android.sourceSets.main.java.srcDirs
         classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
     }
-
-    task dokkaJavadoc(type: dokka.getClass()) {
```
Comment (tuliopereirazup) [inline maven-publish.gradle:84]: This new code do the same thing as the old does?

File: gradle.properties [hunk 5]
```diff
@@ -27,7 +27,7 @@ android.enableSeparateAnnotationProcessing=true
 
 # Publication
 GROUP_NAME=br.com.zup.beagle
-VERSION_NAME=0.0.1
```
Comment (tuliopereirazup) [inline gradle.properties:30]: Back to `0.0.1` version as we don't update this file, the version management will be or using the git tag or a external file in the project.

File: android/beagle/src/test/java/br/com/zup/beagle/android/utils/WidgetExtensionsKtTest.kt [hunk 6]
```diff
@@ -29,20 +29,19 @@ import io.mockk.every
...
     private lateinit var rootView: RootView
 
     @RelaxedMockK
     private lateinit var viewFactoryMock: ViewFactory
 
-    @Before
-    fun setUp() {
+    override fun setUp() {
+        super.setUp()
         MockKAnnotations.init(this)
```
Comment (tuliopereirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/utils/WidgetExtensionsKtTest.kt:45]: As you are using the BaseTest and calling the super method, call for ` MockKAnnotations.init(this)` it's not necessary anymore.

File: android/sample/build.gradle [hunk 7]
```diff
@@ -57,6 +57,9 @@ dependencies {
     implementation AndroidxLibraries.coreKtx
 
     debugImplementation project(Modules.beagleAndroidPreview)
+/*    implementation "br.com.zup.beagle:android:2.2.2"
```
Comment (tuliopereirazup) [inline android/sample/build.gradle:60]: As you are referencing the project and kap bellow, this comment can be removed.

File: android/sample/src/main/java/br/com/zup/beagle/sample/MainActivity.kt [hunk 8]
```diff
@@ -50,6 +52,12 @@ class MainActivity : AppCompatActivity() {
             title = "Beagle Sample"
             elevation = 4.0f.dp()
         }
+/*        startActivity(BeagleActivity.newIntent(
```
Comment (tuliopereirazup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/MainActivity.kt:55]: I think you missed this comment here.

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormInput.kt [hunk 9]
```diff
@@ -37,10 +37,10 @@ import br.com.zup.beagle.core.ServerDrivenComponent
  *                  an UITextField in iOS or any other type of view that can receive and store input from users.
  *
  */
-data class FormInput(
-    val name: String,
-    val required: Boolean? = null,
-    val validator: String? = null,
-    val errorMessage: String? = null,
-    override val child: InputWidget
+open class FormInput(
```
Comment (tuliopereirazup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormInput.kt:40]: Does have any motivation to be open?

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormInputHidden.kt [hunk 10]
```diff
@@ -27,7 +27,7 @@ import br.com.zup.beagle.core.ServerDrivenComponent
  * @param value contain the data you wish to transfer.
  *
  */
-data class FormInputHidden(
-    val name: String,
-    val value: String
+open class FormInputHidden(
```
Comment (tuliopereirazup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormInputHidden.kt:30]: Does have any motivation to be open?

File: backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormSubmit.kt [hunk 11]
```diff
@@ -30,7 +30,7 @@ import br.com.zup.beagle.core.ServerDrivenComponent
  *                  If it is defined as "false" the button will start as "disabled"
  *
  */
-data class FormSubmit(
-    override val child: ServerDrivenComponent,
-    val enabled: Boolean = true
+open class FormSubmit(
```
Comment (tuliopereirazup) [inline backend/backend-widgets/src/main/kotlin/br/com/zup/beagle/widget/form/FormSubmit.kt:33]: Does have any motivation to be open?

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/utils/RoundedImageView.kt [hunk 12]
```diff
@@ -14,19 +14,19 @@
...
 import android.content.Context
 import android.graphics.Canvas
 import androidx.appcompat.widget.AppCompatImageView
-import br.com.zup.beagle.android.utils.applyRadius
 
-internal class BeagleImageView(context: Context) : AppCompatImageView(context) {
-
-    var cornerRadius = 0.0f
+@SuppressLint("ViewConstructor")
+internal class RoundedImageView(context: Context,
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/utils/RoundedImageView.kt:25]: What do you think to use the Kotlin format convention?

```suggestion
internal class RoundedImageView(
    context: Context,
    private val cornerRadius: Double = 0.0
) : AppCompatImageView(context) {
```

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
serialize_ComposeComponent_should_write_the_component_from_build() | br.com.zup.beagle.serialization.jackson.BeagleBuilderSerializerTest | 1.348|
serialize_ScreenBuilder_should_write_the_component_from_build() | br.com.zup.beagle.serialization.jackson.BeagleBuilderSerializerTest | 0.008|


<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
serialize_ComposeComponent_should_write_the_component_from_build() | br.com.zup.beagle.serialization.jackson.BeagleBuilderSerializerTest | 2.977|
serialize_ScreenBuilder_should_write_the_component_from_build() | br.com.zup.beagle.serialization.jackson.BeagleBuilderSerializerTest | 0.032|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: maven-publish.gradle (hunk 4 above)
Comment (uziasferreirazup) [reply maven-publish.gradle:84]: I remove this task because I using the task dokka and dont need create this task more.

**Pull request merged**